app/app/components/modals/Wallet/WalletModal.tsx

```diff
@@ -0,0 +1,190 @@
+                    Detected Wallets
+                  </p>
+                  <div className="flex flex-wrap gap-4">
+                    {detectedWallets.map((detectedWallet) => {
+                      const walletConfig = chainConfig
+                        .flatMap((chain) => chain.wallets)
+                        .find((w) => w.id === detectedWallet.id);
+
+                      console.log(walletConfig);
+
+                      if (!walletConfig) return null;
+
+                      const wallet = {
+                        ...detectedWallet,
+                        icon: walletConfig.icon,
+                      };
+
+                      return (
+                        <Wallet
+                          key={wallet.id}
+                          selectedWallet={selectedWallet}
+                          wallet={wallet}
+                          isSupported={isWalletSupported(wallet)}
+                          setter={setSelectedWallet}
+                        />
+                      );
+                    })}
+                  </div>
+                </div>
+              )}
+              <div>
+                <p className="font-gt-america font-medium text-[12px] text-neutral-600 mb-2">
+                  Other Wallets
+                </p>
+                <div className="flex flex-wrap gap-4">
+                  {chainConfig
+                    .flatMap((chain) => chain.wallets)
+                    .filter(
+                      (wallet, index, self) =>
+                        index === self.findIndex((t) => t.id === wallet.id) &&
+                        !detectedWallets.some((w) => w.id === wallet.id)
+                    )
+                    .map((wallet) => (
+                      <Wallet
+                        key={wallet.id}
+                        selectedWallet={selectedWallet}
+                        wallet={wallet}
+                        isSupported={isWalletSupported(wallet)}
+                        setter={setSelectedWallet}
+                      />
+                    ))}
+                </div>
+              </div>
+            </div>
+            {selectedWallet && (
+              <div className="connect-btn-container mt-6">
+                <UIComponents.Button
+                  label={`Connect ${selectedWallet.name}`}
+                  onClick={handleConnect}
+                  className="w-full"
+                  disabled={!isWalletSupported(selectedWallet)}
+                />
+              </div>
+            )}
+          </div>
+        </Modal>
+      )}
+    </>
+  );
+}
+
+const Wallet = ({
+  selectedWallet,
+  wallet,
+  isSupported,
+  setter,
+}: {
+  selectedWallet: WalletOption | null;
+  wallet: WalletOption;
+  isSupported: boolean;
+  setter: (wallet: WalletOption) => void;
+}) => (
+  <button
+    className={`bg-white p-[12px] flex gap-4 items-center border-[2px] rounded-2xl transition-all duration-75 ${
```

Comment:
1. do not hardcode px values
2. use twmerge

this applies to nearly all files.